core/templates/domain/recommendations/exploration-recommendations-backend-api.service.ts

```diff
@@ -52,8 +52,11 @@ export class ExplorationRecommendationsBackendApiService {
   async getRecommendedSummaryDictsAsync(
       authorRecommendedExpIds: string[],
       includeSystemRecommendations: string,
-      collectionId: string, storyId: string, currentNodeId: string,
-      explorationId: string): Promise<LearnerExplorationSummary[]> {
+      // 'collectionId', 'storyId' and 'currentNodeId' can be null
+      // since they may not be present in the URL.
+      collectionId: string | null, storyId: string | null,
+      currentNodeId: string | null, explorationId: string):
+      Promise<LearnerExplorationSummary[]> {
```

Comment:
```suggestion
      // 'collectionId', 'storyId' and 'currentNodeId' can be null
      // since they may not be present in the URL.
      collectionId: string | null, 
      storyId: string | null,
      currentNodeId: string | null, 
      explorationId: string
  ): Promise<LearnerExplorationSummary[]> {
```

Reply:
done